andela/ah-backend-realers
# 166187865 favorite or unfavorite an article

#### What does this PR do?
- enable an authenticated user to favorite or unfavorite an article

#### Description of Task to be completed?

- add functionality to favorite an article
- add functionality to unfavorite an article
- add unit tests to the added functionalities

#### How should this be manually tested?
- fetch this repo locally in the terminal with `$ git fetch origin ft-favorite-articles-166187865`
- run `$ python manage.py migrate articles`
- Now test out the following.
   1. Favorite an article with this `POST ​/api/articles​/{slug}​/favorite​/` with no request body
   2. Unfavorite an article with `DELETE /api​/articles​/{slug}​/favorite​/`
   3. Retrieve all user's favorite articles with `GET /api/favorite-articles/`
   4. Do `GET /api/articles/{slug}/` for the favorited articles to confirm that the response body contains the `is_favorited` set to **True**
  5. Kindly confirm that the rest of the functionalities are still working well.

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?

- Story ID - [166187865](https://www.pivotaltracker.com/story/show/166187865)
